### DISCUSSION TITLE
A function that returns a populated `Resources` struct

### DISCUSSION THREAD

**Discussion by iholder101** [opencontainers/runc#4414] [Ideas]
**The problem:**
AFAICT, currently there is no way of getting a populated `Resources` struct from the cgroup manager.

This means that in order to understand the current configuration of a cgroup, there's a need to:
* Call `manager.GetPaths()`.
* Iterate on the given paths for every different controller/subsystem.
* For every subsystem: read the different files, parse them, etc.

The above should be written in two versions to support both cgroup v1 and v2.

**The solution:**
I would expect that the manager would support a `GetResources()` function that returns a populated `Resources` struct. This seems like a very valuable action that has to be re-implemented in many different projects.

If this idea would be accepted, I'm willing to invest time in implementing it.

p.s. the `manager.GetStats()` method is not enough, since it can't, as a single example, get `cpu.wight`'s value.

**Comment by lifubang** [comment]
Though we want to move most packages into internal, but I think ‘libct/cgroups’ is not included. So SGTM, please feel free to open a PR to implement this.
